server/src/internalClusterTest/java/org/elasticsearch/index/shard/SearchIdleIT.java

```diff
@@ -175,7 +182,9 @@ public void testPendingRefreshWithIntervalChange() throws Exception {
         // otherwise, it will compete to call `Engine#maybeRefresh` with the `scheduledRefresh` that we are going to verify.
         ensureNoPendingScheduledRefresh(indexService.getThreadPool());
         client().prepareIndex("test").setId("2").setSource("{\"foo\" : \"bar\"}", XContentType.JSON).get();
-        assertTrue(shard.scheduledRefresh());
+        PlainActionFuture<Boolean> future3 = PlainActionFuture.newFuture();
```

Comment:
This pattern looks repetitive. Maybe some helper function would help refactoring them in an one-liner?